lib/beaker/hypervisor/openstack.rb

```diff
@@ -27,25 +28,36 @@ def initialize(openstack_hosts, options)
       @hosts = openstack_hosts
       @vms = []
 
-      raise 'You must specify an Openstack API key (:oopenstack_api_key) for OpenStack instances!' unless @options[:openstack_api_key]
+      raise 'You must specify an Openstack API key (:openstack_api_key) for OpenStack instances!' unless @options[:openstack_api_key]
       raise 'You must specify an Openstack username (:openstack_username) for OpenStack instances!' unless @options[:openstack_username]
       raise 'You must specify an Openstack auth URL (:openstack_auth_url) for OpenStack instances!' unless @options[:openstack_auth_url]
       raise 'You must specify an Openstack tenant (:openstack_tenant) for OpenStack instances!' unless @options[:openstack_tenant]
       raise 'You must specify an Openstack network (:openstack_network) for OpenStack instances!' unless @options[:openstack_network]
-      @compute_client ||= Fog::Compute.new(:provider => :openstack,
-                                           :openstack_api_key => @options[:openstack_api_key],
-                                           :openstack_username => @options[:openstack_username],
-                                           :openstack_auth_url => @options[:openstack_auth_url],
-                                           :openstack_tenant => @options[:openstack_tenant])
+
+      optionhash = {}
+      optionhash[:provider]           = :openstack
+      optionhash[:openstack_api_key]  = @options[:openstack_api_key]
+      optionhash[:openstack_username] = @options[:openstack_username]
+      optionhash[:openstack_auth_url] = @options[:openstack_auth_url]
+      optionhash[:openstack_tenant]   = @options[:openstack_tenant]
+      optionhash[:openstack_region]   = @options[:openstack_region] if @options[:openstack_region]
+
+      @compute_client ||= Fog::Compute.new(optionhash)
+
       if not @compute_client
         raise "Unable to create OpenStack Compute instance (api key: #{@options[:openstack_api_key]}, username: #{@options[:openstack_username]}, auth_url: #{@options[:openstack_auth_url]}, tenant: #{@options[:openstack_tenant]})"
       end
-      @network_client ||= Fog::Network.new(
-        :provider => :openstack,
-        :openstack_api_key => @options[:openstack_api_key],
-        :openstack_username => @options[:openstack_username],
-        :openstack_auth_url => @options[:openstack_auth_url],
-        :openstack_tenant => @options[:openstack_tenant])
+
+      networkoptionhash = {}
+      networkoptionhash[:provider]           = :openstack
+      networkoptionhash[:openstack_api_key]  = @options[:openstack_api_key]
+      networkoptionhash[:openstack_username] = @options[:openstack_username]
+      networkoptionhash[:openstack_auth_url] = @options[:openstack_auth_url]
+      networkoptionhash[:openstack_tenant]   = @options[:openstack_tenant])
+      networkoptionhash[:openstack_region]   = @options[:openstack_region] if @options[:openstack_region],
```

Comment:
unnecessary line-ending comma here.  copy-paste error?  Otherwise, everything looks clean.  Once this is fixed, I'll be happy to push this through last round testing and getting it merged.